crates/librqbit/src/storage/examples/mmap.rs

```diff
@@ -63,11 +63,15 @@ impl TorrentStorage for MmapStorage {
         anyhow::bail!("not implemented")
     }
 
-    fn init(&mut self, _meta: &ManagedTorrentInfo) -> anyhow::Result<()> {
+    fn init(&mut self, _meta: &ManagedTorrentShared) -> anyhow::Result<()> {
         Ok(())
     }
 
     fn remove_directory_if_empty(&self, _path: &std::path::Path) -> anyhow::Result<()> {
         Ok(())
     }
+
+    fn on_piece_completed(&self, _file_id: usize, _offset: u64) -> anyhow::Result<()> {
```

Comment:
It would probably make sense to implement default in the trait itself not to add this to every single storage example that exists.

There's also "storage_middleware" feature that will break if turned on (although I do understand it will break already anyway because ManagedTorrentInfo was renamed, but that's a separate thing)

Reply:
Added default (empty) implementation in the trait itself, can still be override in trait implementations.

I checked and a `cargo check --all-features` build (apart from an error of multiple definition of Sha1

```
error[E0428]: the name `Sha1` is defined multiple times
  --> crates/sha1w/src/lib.rs:79:1
   |
76 | pub type Sha1 = crypto_hash_impl::Sha1CryptoHash;
   | ------------------------------------------------- previous definition of the type `Sha1` here
...
79 | pub type Sha1 = ring_impl::Sha1Ring;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Sha1` redefined here
   |
   = note: `Sha1` must be defined only once in the type namespace of this module
```